libipl/libipl.C

```diff
@@ -366,3 +366,8 @@ void ipl_set_apply_guards(bool apply)
 {
     g_ipl_apply_guards = apply;
 }
+
+bool ipl_apply_guards()
```

Comment:
guard_policy is a bit of a generic name i.e that policy flag is also used to prevent the guard records creation/deletion but, in libipl we are just controlling guard apply logic based on that flag. 

If you want to use as guard_policy from a futuristic point of view then I can change i.e if this policy flag can be also used for someother purpose in libipl. AFAIK - we will only control the apply guard logic.

Reply:
Let's avoid adding more global flags.  Instead let's define a state or a context (as a structure) which holds all the settings information.  In that case you just need a single private api to get the state (or context) and can use the elements directly.

Reply:
Introduced the `ipl_settings` context.